yamux/src/connection/stream.rs

```diff
@@ -218,7 +218,8 @@ impl Stream {
             drop(shared);
 
             let mut frame = Frame::window_update(self.id, credit).right();
-            self.add_flag(frame.header_mut());
+            let mut parsed_frame = frame.parse_mut().expect("valid frame");
+            self.add_flag(parsed_frame.header_mut());
```

Comment:
Can we make this more ergonomic somehow? It is duplicated with line 376.

Reply:
ok add_flag() now takes a Frame and parses it and update the frame's flags